test/functionalTest/cases/0501_cors/allowed_headers.test

```diff
@@ -0,0 +1,72 @@
+# Copyright 2014 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+CORS allowed headers
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0-255 IPV4 -cors __ALL
+
+--SHELL--
+echo "01. Allowed headers: GET OPTIONS"
+echo "=============================================================================="
+orionCurl --url /v2 -X OPTIONS
+echo
+echo
+
+echo "02. Allowed headers: GET POST OPTIONS"
+echo "=============================================================================="
+orionCurl --url /v2/entities -X OPTIONS
+echo
+echo
+
+--REGEXPECT--
+01. Allowed headers: GET OPTIONS
+==============================================================================
+HTTP/1.1 200 OK
+Content-Length: 0
+Access-Control-Max-Age: REGEX([0-9]{5})
+Access-Control-Allow-Headers: REGEX(.*)
+Access-Control-Allow-Origin: *
+Access-Control-Allow-Methods: GET, OPTIONS

```

Comment:
According to CORS specification the OPTIONS verb that is used to get the list of methods in this header should be also included in the list, isn't it?

Reply:
I am sorry I didn't get this. OPTIONS is already listed in the Access-Control-Allow-Methods list. Can you please elaborate?

Reply:
It is only a doubt... not sure if OPTIONS should be there or no. My doubt is based in my lack of expersie on CORS.

However, I understand that if you haven't get this is probably because the answer to my doubt is "yes, OPTIONS has to be there" :)

NTC

Reply:
Ah ok, I see your point. Well this was a question mark for me as well during the implementation. The examples in the spec add OPTIONS to the responses and also I checked the REST APIs of Google, Github etc. and it is always there :)